aws-sdk-go-v2/Dockerfile

```diff
@@ -0,0 +1,35 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+ARG GO_VERSION
+FROM public.ecr.aws/docker/library/golang:${GO_VERSION}
+
+ENV XDG_CACHE_HOME /tmp
+
+RUN go env -w GOPROXY=direct
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends openjdk-17-jdk
```

Comment:
JDK 17 is pretty old at this point, since this is a new dockerfile is there any reason we cant use something newer? Would jdk-23 work?

Reply:
Updated it to install OpenJDK 23 via Amazon Coretto in https://github.com/aws/amazon-ecs-agent/pull/4406/commits/9685bbfe24b415cffa8a2978a467e6e619b1b70a (the base image is Debian Bookworm, which doesn't support anything newer than OpenJDK 17).